libraries/griptape_nodes_library/griptape_nodes_library/config/prompt/griptape_cloud_prompt.py

```
@@ -120,3 +124,27 @@ def validate_before_workflow_run(self) -> list[Exception] | None:
             api_key_env_var=API_KEY_ENV_VAR,
             api_key_url=API_KEY_URL,
         )
+
+    def _list_models(self) -> tuple[list[str], str]:
+        """Returns the list of available models from Griptape Cloud, and the default model.
+
+        This method fetches the list of models from the Griptape Cloud API and
+        returns them. If the API call fails, it falls back to the default list
+        of models defined in the `MODEL_CHOICES` constant.
+
+        Returns:
+            tuple: A tuple containing a list of available model names and the default model name.
+        """
+        # Fetch the list of available models from the Griptape Cloud API.
+        response = requests.get(
+            CHAT_MODELS_URL,
+            headers={"Authorization": f"Bearer {self.get_config_value(service=SERVICE, value=API_KEY_ENV_VAR)}"},
+            timeout=10,
+        )
+        try:
+            response.raise_for_status()
+            models = response.json()["models"]
+            return [model["model_name"] for model in models], next(filter(lambda x: x["default"], models))["model_name"]
+
+        except requests.HTTPError:
+            return MODEL_CHOICES, DEFAULT_MODEL
```

we should let this raise an exception instead of papering over the issue.

---

I think it's reasonable to default here, though we should definitely log that an error occurred.